plugin/src/main/java/com/playmonumenta/scriptedquests/zones/zone/ZoneFragment.java

```diff
@@ -0,0 +1,254 @@
+			upperMin.setX(pos.getX());
+			break;
+		case Z:
+			lowerMax.setZ(pos.getZ());
+			upperMin.setZ(pos.getZ());
+			break;
+		default:
+			lowerMax.setY(pos.getY());
+			upperMin.setY(pos.getY());
+		}
+
+		lower.trueMaxCorner(lowerMax);
+		upper.minCorner(upperMin);
+
+		result[0] = lower;
+		result[1] = upper;
+
+		return result;
+	}
+
+	/*
+	 * Returns a list of fragments of this zone, split by an overlapping zone.
+	 * Does not include overlap or register a new parent.
+	 */
+	public ArrayList<ZoneFragment> splitByOverlap(BaseZone overlap) {
+		return splitByOverlap(overlap, null);
+	}
+
+	/*
+	 * Returns a list of fragments of this zone, split by an overlapping zone.
+	 * Optionally register a new parent and return the center zone.
+	 *
+	 * When registering a new parent, only do so for one of the parent zones.
+	 * The other parent zone should have the overlap removed as normal to avoid
+	 * overlapping fragments.
+	 */
+	public ArrayList<ZoneFragment> splitByOverlap(BaseZone overlap, Zone newParent) {
+		ZoneFragment centerZone = new ZoneFragment(this);
+
+		Vector otherMin = overlap.minCorner();
+		Vector otherMax = overlap.trueMaxCorner();
+
+		ZoneFragment[] tempSplitResult;
+		ArrayList<ZoneFragment> result = new ArrayList<ZoneFragment>();
+
+		for (Axis axis : Axis.values()) {
+			ZoneFragment lower;
+			ZoneFragment upper;
+
+			ArrayList<ZoneFragment> workZones = result;
+			result = new ArrayList<ZoneFragment>();
+
+			for (ZoneFragment workZone : workZones) {
+				// Add zones split from existing split zones
+				tempSplitResult = workZone.splitAxis(otherMin, axis);
+				lower = tempSplitResult[0];
+				workZone = tempSplitResult[1];
+
+				tempSplitResult = workZone.splitAxis(otherMax, axis);
+				workZone = tempSplitResult[0];
+				upper = tempSplitResult[1];
+
+				if (lower != null && lower.isValid()) {
+					result.add(lower);
+				}
+				if (workZone != null && workZone.isValid()) {
+					result.add(workZone);
+				}
+				if (upper != null && upper.isValid()) {
+					result.add(upper);
+				}
+			}
+
+			// Add zones split from center, but not the center (overlap) itself
+			tempSplitResult = centerZone.splitAxis(otherMin, axis);
+			lower = tempSplitResult[0];
+			centerZone = tempSplitResult[1];
+
+			tempSplitResult = centerZone.splitAxis(otherMax, axis);
+			centerZone = tempSplitResult[0];
+			upper = tempSplitResult[1];
+
+			if (lower != null && lower.isValid()) {
+				result.add(lower);
+			}
+			if (upper != null && upper.isValid()) {
+				result.add(upper);
+			}
+		}
+
+		// If registering a new parent, it may be added now that the center zone is the size of the overlap.
+		if (newParent != null) {
+			centerZone.mParents.put(newParent.getLayerName(), newParent);
+			result.add(centerZone);
+		}
+
+		return result;
+	}
+
+	/*
+	 * Merge two ZoneFragments without changing their combined size/shape.
+	 *
+	 * Returns the merged ZoneFragment or None.
+	 */
+	public ZoneFragment merge(ZoneFragment other) {
```

Comment:
I assume this is used for defragment? (doesn't seem to be used yet)

Reply:
Yes, it's used for defragmenting. This can reduce the number of fragments in Region 1 and Region 2 to 1/2 to 1/3 of what they are without defragmenting.